### PR TITLE
Chore: removing useEndpointData from license api

### PR DESCRIPTION
**Pull request by hugocostadev**
## Proposed changes (including videos or screenshots)

The deprecated hook `useEndpointData` was removed and changed to `useMethod` + `useQuery`separated in a hook.

In `useQuery` options we added an option to keep the cache as `Infinity` until the queries are invalidated or the page refreshes, also the query keys were changed to consolidate the keys and manage the cache in the best way possible to avoid unnecessary API calls. 

In settings changed or user action that impact the License status was added a method to invalidate the cache for license query keys

E.g: `queryClient.invalidateQueries(['licenses']);`

### CODE REVIEW

File: apps/meteor/client/hooks/useIsEnterprise.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import { useEndpoint } from '@rocket.chat/ui-contexts';
 import { useQuery } from '@tanstack/react-query';
 
-export const useIsEnterprise = (): boolean => {
+export const useIsEnterprise = (): { isEnterprise: boolean; isLoading: boolean } => {
```
Comment (gabriellsh) [inline apps/meteor/client/hooks/useIsEnterprise.ts:4]: Please, don't hide the return of `useQuery`. Always return all the properties. There's a helper type from `@tanstack/react-query` called `UseQueryResult`, that you can use to wrap the type that comes in `data`. See `apps/meteor/client/hooks/useRoomInfoEndpoint.ts` for more context.

File: apps/meteor/client/hooks/useLicense.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { CloudRegistrationStatus, ILicense } from '@rocket.chat/core-typings';
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+
+export const useLicense = (): {
```
Comment (gabriellsh) [inline apps/meteor/client/hooks/useLicense.ts:5]: As we talked on a DM, move this to 2 separate hooks.

File: apps/meteor/client/hooks/useLicense.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	licenses: ILicense[] | undefined;
+	registrationStatus: CloudRegistrationStatus | undefined;
+	isError: boolean;
+	isLoading: boolean;
+	isSuccess: boolean;
+} => {
+	const getRegistrationStatus = useEndpoint('GET', '/v1/cloud.registrationStatus');
+	const getLicenses = useEndpoint('GET', '/v1/licenses.get');
+
+	const { data: registrationStatusData } = useQuery(['licenses', 'getRegistrationStatus'], () => getRegistrationStatus(), {
```
Comment (gabriellsh) [inline apps/meteor/client/hooks/useLicense.ts:15]: `registrationStatus` and `licenses` are somewhat related, but at the same time independent from each other. Can you change this key? Thanks.

**Review by gabriellsh — CHANGES_REQUESTED** [review]

File: apps/meteor/client/hooks/useIsEnterprise.ts (hunk 1 above)
Comment (hugocostadev) [reply apps/meteor/client/hooks/useIsEnterprise.ts]: Done!

File: apps/meteor/client/hooks/useLicense.ts (hunk 3 above)
Comment (hugocostadev) [reply apps/meteor/client/hooks/useLicense.ts]: Done!

File: apps/meteor/client/hooks/useLicense.ts (hunk 2 above)
Comment (hugocostadev) [reply apps/meteor/client/hooks/useLicense.ts]: Done!

**Review by gabriellsh — APPROVED** [review]

**Pull request merged**